docs/content/TreeView.mdx

```diff
@@ -104,6 +108,92 @@ description: A hierarchical list of items where nested items can be expanded and
 </Box>
 ```
 
+### With React Router
+
+```jsx
+import {Link, useMatch, useResolvedPath, navigate} from 'react-router-dom'
+import {TreeView} from '@primer/react'
+
+function LinkItem({id, to, children}) {
+  const resolved = useResolvedPath(to)
+  const isCurrent = useMatch({path: resolved.pathname, end: true})
+  return (
+    <TreeList.Item
+      id={id}
+      aria-current={isCurrent ? 'page' : undefined}
+      current={isCurrent}
+      onSelect={() => navigate(to)}
+    >
+      {children}
+    </TreeList.Item>
+  )
+}
+
+function App() {
+  return (
+    <Box sx={{display: 'grid', gap: 2, maxWidth: 400}}>
+      <nav aria-label="Files">
+        <TreeList aria-label="Files">
+          <LinkItem id="src/Avatar.tsx" to="/src/Avatar.tsx">
+            Avatar.tsx
+          </LinkItem>
+          <LinkItem id="src/Button.tsx" to="/src/Button.tsx">
+            Button.tsx
+          </LinkItem>
+        </TreeList>
+      </nav>
+    </Box>
+  )
+}
+```
+
+### With Next.js
+
+```jsx
+import {useRouter} from 'next/router'
+import Link from 'next/link'
+import {TreeView} from '@primer/react'
+
+function LinkItem({id, href, children}) {
+  const router = useRouter()
+  const isCurrent = typeof href === 'string' ? router.asPath === href : router.pathname === href.pathname
+  return (
+    <Link href={href} passHref>
+      <TreeView.Item
+        id={id}
+        aria-current={isCurrent ? 'page' : false}
+        current={isCurrent}
+      >
+        {children}
+      </TreeView.Item>
+    </Link>
+  )
+}
+
+function App() {
+  return (
+    <Box sx={{display: 'grid', gap: 2, maxWidth: 400}}>
+      <nav aria-label="Files">
+        <TreeList aria-label="Files">
+          <LinkItem id="src/Avatar.tsx" href="/src/Avatar.tsx">
+            Avatar.tsx
+          </LinkItem>
+          <LinkItem 
+            id="${component}.${filetype}" 
+            href={{ 
+              pathname: "/src/[component].[filetype]",
+              query: { component, impl }
+            }}
+          >
+            {"${component}.${filetype}"}
+          </LinkItem>
```

Comment:
I wonder if this will be confusing because `component` and `impl` aren't defined. What about something like this?
```suggestion
          <LinkItem 
            id="src/Button.tsx"
            // You can also pass a URL object
            // https://nextjs.org/docs/api-reference/next/link#with-url-object
            href={{ 
              pathname: "/src/[component].[filetype]",
              query: { component: 'Button', filetype: 'tsx' }
            }}
          >
            Button.tsx
          </LinkItem>
```